examples/webgl_materials_reprojection.html

```diff
@@ -0,0 +1,243 @@
+		</div>
+
+		<!-- Import maps polyfill -->
+		<!-- Remove this when import maps will be widely supported -->
+		<script async src="https://unpkg.com/es-module-shims@1.6.3/dist/es-module-shims.js"></script>
+
+		<script type="importmap">
+			{
+				"imports": {
+					"three": "../build/three.module.js",
+					"three/addons/": "./jsm/"
+				}
+			}
+		</script>
+
+		<script type="module" >
+
+			import * as THREE from 'three';
+
+			import { OrbitControls } from 'three/addons/controls/OrbitControls.js';
+			import { RGBELoader } from 'three/addons/loaders/RGBELoader.js';
+			import * as CameraUtils from 'three/addons/utils/CameraUtils.js';
+
+			import { GUI } from 'three/addons/libs/lil-gui.module.min.js';
+			import Stats from 'three/addons/libs/stats.module.js';
+
+			let camera, scene, renderer, stats;
+			let cube, curvedPanel, material;
+
+			let eyeCamera, eyeRenderTarget, eyeCameraHelper, params = { setFoVToCorners: true, panelCurvature: 20 };
+
+			let controls;
+
+			init();
+
+			function init() {
+
+				renderer = new THREE.WebGLRenderer({ antialias: true });
+				renderer.setPixelRatio(window.devicePixelRatio);
+				renderer.setSize(window.innerWidth, window.innerHeight);
+				renderer.setAnimationLoop(animation);
+				renderer.outputEncoding = THREE.sRGBEncoding;
+				renderer.toneMapping = THREE.ACESFilmicToneMapping;
+				document.body.appendChild(renderer.domElement);
+
+				window.addEventListener('resize', onWindowResized);
+
+				stats = new Stats();
+				document.body.appendChild(stats.dom);
+
+				scene = new THREE.Scene();
+
+				camera = new THREE.PerspectiveCamera(60, window.innerWidth / window.innerHeight, 1, 1000);
+				camera.position.x = 60;
+				camera.position.y = 20;
+				camera.position.z = 30;
+				scene.add(camera);
+
+				new RGBELoader()
+					.setPath('textures/equirectangular/')
+					.load('quarry_01_1k.hdr', function (texture) {
+						texture.mapping = THREE.EquirectangularReflectionMapping;
+						scene.background = texture;
+						scene.environment = texture;
+					});
+
+				//
+
+				eyeRenderTarget = new THREE.WebGLRenderTarget(1024, 576);
+				eyeRenderTarget.texture.type = THREE.HalfFloatType;
+				eyeRenderTarget.texture.encoding = renderer.outputEncoding;
+
+				eyeCamera = new THREE.PerspectiveCamera(30, 16.0 / 9.0, 1, 100);
+				scene.add(eyeCamera);
+				eyeCameraHelper = new THREE.CameraHelper(eyeCamera);
+				scene.add(eyeCameraHelper);
+
+				//
+
+				material = new THREE.MeshBasicMaterial({ map: eyeRenderTarget.texture });
+				material.uniforms = {
+					eyeProjectionMatrix : { value: eyeCamera.projectionMatrix },
+					eyeViewMatrix       : { value: eyeCamera.matrixWorldInverse },
+					previewOnScreen        : { value: false }
+				};
+
+				material.onBeforeCompile = (shader) => {
+
+					// Vertex Shader: Set UVs to be eye-camera screen-space; Map positions to be main screen-space.
+					shader.vertexShader =
+						shader.vertexShader.slice( 0, 18 ) +
```

Comment:
I'm afraid such constructions can break easily.

Reply:
Could you also implement the shader with a custom `ShaderMaterial`? The material itself does not have to support any features other then the reprojection and diffuse map logic.

Reply:
Ah; that's a fair point; I was a little worried about the tonemapping and encoding steps, but I found the right macros for them.